server/models/idea.js

```diff
@@ -0,0 +1,45 @@
+const MongoClient = require('mongodb').MongoClient;
+const url = 'mongodb://localhost:27017/rankerdb';
+
+exports.all = function(callback) {
+	MongoClient.connect(url, function(err, db) {
+	  if(err)throw(err);
+
+  	db.collection('ideas').find().toArray(function(err, result) {
+  		if (err) throw err;
```

Comment:
When you use callbacks, you can't really throw errors like this since the original call stack isn't there to handle any errors that might occur. Basically, the way to deal with errors is to pass them as the first parameter in the callback like the `toArray` function is doing

https://docs.nodejitsu.com/articles/errors/what-are-the-error-conventions/